test/common/storagelib.py

```diff
@@ -89,7 +89,9 @@ def add_loopback_disk(self, size=50, name=None):
         # them.
         self.machine.execute("partprobe '%s'" % dev)
         # right after unmounting the device is often still busy, so retry a few times
-        self.addCleanup(self.machine.execute, f"umount {dev} || true; rm $(losetup -n -O BACK-FILE -l {dev}); until losetup -d {dev}; do sleep 1; done", timeout=10)
+        self.addCleanup(self.machine.execute, f"""for device in "$(findmnt -n {dev} | sed 's@ {dev}.*@@')"; do umount $device; done ; rm $(losetup -n -O BACK-FILE -l {dev}); until losetup -d {dev}; do sleep 1; done""", timeout=10)
```

Comment:
`findmnt --noheadings --output=TARGET {dev}` please, that avoids the seddery and is more robust. You can also shorten this a bit to

```
findmnt --noheadings --output=TARGET {dev}` | xargs umount
```

Also, please split this into two lines -- the line is too long, and the commands are also sufficiently complex so that it makes sense to run these as two separate commands -- makes it easier to see which part fails.

Reply:
thanks for suggestion, seems well.

Reply:
**update**: this solution does not work well, if there is no mount point for device found the command pipe  fails. have to add for cycle back.

Reply:
Ah right, you need `xargs --no-run-if-empty`